juniper_junos-formula/bin/compile_junos_data.py

```diff
@@ -0,0 +1,371 @@
+            switch_pillar[device]['ignore_ports'] += switch_ignore
+            # why this set?
+            switch_pillar[device]['ignore_ports'] = list(set(switch_pillar[device]['ignore_ports']))
+
+    ## Compatibility helper for the old, list based, input format
+    data_vlans = data.get('vlans')
+    if isinstance(data_vlans, list):
+        vlans = {}
+        for vlan in data_vlans:
+            vlan_id = vlan.get('id')
+            vlan_name = vlan.get('name')
+            vlan_description = vlan.get('description')
+            vlan_groups = vlan.get('groups')
+            vlans.update({
+                vlan_name: {
+                    'id': vlan_id,
+                    'description': vlan_description
+                }
+            })
+            if vlan_groups:
+                vlans[vlan_name].update({'groups': vlan_groups})
+        logger.debug(f'Converted legacy VLANs: {vlans}')
+    elif isinstance(data_vlans, dict):
+        vlans = data_vlans
+
+    for vlan, vconfig in vlans.items():
+        vlan_id = vconfig.get('id')
+        vlan_description = vconfig.get('description')
+
+        logger.debug(f'Processing VLAN {vlan} with ID {vlan_id}')
+
+        for group in vconfig.get('groups', []):
+            logger.debug(f'Processing group {group}')
+
+            if group in global_port_groups:
+                for group, gconfig in global_port_groups[group].items():
+                    group_description = gconfig.get('description')
+                    group_members = gconfig.get('members', [])
+
+                    if vlan_id in switch_pillar[group]['vlan_set']:
+                        logger.debug(f'VLAN {vlan} already exists in set.')
+                    else:
+                        # ??
+                        switch_pillar[group]['vlan_set'] = switch_pillar[group]['vlan_set'].union({vlan_id})
+                        switch_pillar[group]['vlans'].update({vlan: {'id': vlan_id, 'description': vlan_description}})
+
+                    for port in group_members:
+                        if not 'iface' in port:
+                            logger.debug(f'Skipping {port}')
+                            continue
+
+                        interface = port['iface']
+
+                        if interface in switch_pillar[group]['ignore_ports']:
+                            logger.error(f'This should not happen?')
+                            sys.exit(1)
+
+                        interface_description = port.get('description')
+
+                        interface_pillar = switch_pillar[group]['ports']
+                        if interface in interface_pillar:
+                            logger.debug(f'Interface {interface} is already in pillar')
+                            interface_pillar = switch_pillar[group]['ports'][interface]
+                            tagged = interface_pillar['tagged']
+                            if vlan_id not in tagged:
+                                tagged.append(vlan_id)
```

Comment:
It is somehow implicitly used, through this assignment

```
interface_pillar = switch_pillar[group]['ports']
``` 

it is later on written to the output files under `ports:<interface/port name>:tagged:<list of tagged interfaces>`. If I remove it there are still some list entries, but less.

Reply:
Hmm, maybe you are looking at something different than I do ;-)

I assume what you see in the output file comes from `'tagged': [vlan_id],` a few lines below this comment.

What I meant is that the variable `tagged´. It gets set in
```
                           tagged = switch_pillar[group]['ports'][interface]['tagged']
                            if vlan_id not in tagged:
                                tagged.append(vlan_id)
```
but never gets used, which also means these lines are unused/superfluous. (If you really removed them and got a different output, then something really strange is happening.)

(And I still wonder if removing the 3 lines is the correct solution, or if you should keep them and change `'tagged': [vlan_id],` to `'tagged': tagged,` a few lines below this comment. In practise, this would mean that vlan_id would still be there, but possibly also other values that pre-existed in `switch_pillar[group]['ports'][interface]['tagged']`.)

Reply:
From my understanding it does not get _explicitly_ used as a variable, but it gets used as a pointer reference to `switch_pillar[group]['ports'][interface]['tagged']`.

Reply:
Indeed, that explains it :-)

Feel free to close this thread as invalid ;-)  (and optionally add a comment to the code for future readers)

Reply:
Glad to hear it's nothing more peculiar!